src/adapter.ts

```diff
@@ -353,19 +367,88 @@ export class CeedlingAdapter implements TestAdapter {
+        let filePrefix = 'test_';
+        if (ymlProjectData) {
+            try {
+                const ymlProjectTestPrefix = ymlProjectData[':project'][':test_file_prefix'];
+                if (ymlProjectTestPrefix != undefined) {
+                    filePrefix = ymlProjectTestPrefix;
+                }
+            } catch (e) { }
+        }
+        this.fileLabelRegex = new RegExp(`.*\/${filePrefix}(.*).c`, 'i');
+    }
+
+    private getFileLabelRegex(): RegExp {
+        if (!this.fileLabelRegex) {
+            this.setFileLabelRegex();
+        }
+        return this.fileLabelRegex as RegExp;
+    }
+
+    private setTestLabelRegex(ymlProjectData: any = undefined) {
+        let testPrefix = 'test|spec|should';
+        if (ymlProjectData) {
+            try {
+                const ymlProjectTestPrefix = ymlProjectData[':unity'][':test_prefix'];
+                if (ymlProjectTestPrefix != undefined) {
+                    testPrefix = ymlProjectTestPrefix;
+                }
+            } catch (e) { }
+        }
+        this.testLabelRegex = new RegExp(`(?:${testPrefix})_*(.*)`);
+    }
+
+    private getTestLabelRegex(): RegExp {
+        if (!this.testLabelRegex) {
+            this.setTestLabelRegex();
+        }
+        return this.testLabelRegex as RegExp;
+    }
+
+    private setTestLabel(testName: string): string | undefined{
+        let testLabel = testName;
+        if (this.isPrettyTestLabelEnable) {
+            const labelFunctionRegex = this.getTestLabelRegex();
+            let testLabelMatches = labelFunctionRegex.exec(testName);
+            if (testLabelMatches != null) {
+                testLabel = testLabelMatches[1];
+            }
+        }
+        return testLabel;
+    }
+
+    private setFileLabel(fileName : string): string {
+        let fileLabel = fileName;
+        if (this.isPrettyTestFileLabelEnable) {
+            const labelFileRegex = this.getFileLabelRegex();
+            let labelMatches = labelFileRegex.exec(fileName);
+            if (labelMatches != null) {
+                fileLabel = labelMatches[1];
+            }
+        }
+        return fileLabel;
+    }
+
     private setTestSuiteInfo(files: string[]) {
         this.testSuiteInfo = {
             type: 'suite',
             id: 'root',
             label: 'Ceedling',
             children: []
         } as TestSuiteInfo;
+        /* get labels configuration */
+        this.isPrettyTestFileLabelEnable = this.getConfiguration().get<boolean>('prettyTestFileLabel', false);
+        this.isPrettyTestLabelEnable = this.getConfiguration().get<boolean>('prettyTestLabel', false);
+        
         for (const file of files) {
             const fullPath = path.resolve(this.getProjectPath(), file);
+            const fileLabel = this.setFileLabel(file);
             const currentTestSuitInfo: TestSuiteInfo = {
                 type: 'suite',
                 id: file,
-                label: file,
+                label: fileLabel,
```

Comment:
This field is also used by [src/adapter.ts:620](https://github.com/numaru/vscode-ceedling-test-adapter/blob/44cda198c5e8aed72bfeb59d0585e07cde98a8ba/src/adapter.ts#L620) to select which test suite to run. But the`ceedling test:*` [command](https://github.com/ThrowTheSwitch/Ceedling/blob/master/docs/CeedlingPacket.md#now-what-how-do-i-make-it-go) accept test module regexps as well as file names.

I'm trying to see if it can introduce some side effects.

I think we can exclude the name collision `test/a/testFoo` and `test/b/testFoo`. The user cannot know which test he decided to start.

There is another possible collision: `test/testFoo` and `test/testFooAndBar`. I was expecting the start of `testFoo` to also run `testFooAndBar` but ... I did the test and it is not the case.

So, it looks good to me! 👍